cmd/av/stack_tree.go

```diff
@@ -68,14 +68,85 @@ func printStackTree(repo *git.Repo, branches map[string]meta.Branch, currentBran
 		fmt.Printf("%s<ERROR: unknown branch: %s>\n", indent, root)
 		return
 	}
+
+	branchInfo := getBranchInfo(repo, branch)
+
 	if currentBranch == branch.Name {
 		_, _ = fmt.Print(
-			indent, colors.Success("* "), colors.Success(branch.Name), "\n",
+			indent, colors.Success("* "), colors.Success(branch.Name), " ", colors.Faint(branchInfo), "\n",
 		)
 	} else {
-		_, _ = fmt.Printf("%s%s\n", indent, branch.Name)
+		_, _ = fmt.Print(indent, branch.Name, " ", colors.Faint(branchInfo), "\n")
 	}
 	for _, next := range branch.Children {
 		printStackTree(repo, branches, currentBranch, next, depth+1)
 	}
 }
+
+func getBranchInfo(repo *git.Repo, branch meta.Branch) string {
+	var branchInfo string
+
+	parentStatus, err := getParentStatus(repo, branch)
+	if err != nil {
+		fmt.Printf("<ERROR: could not get status of parent branch: %v>\n", err)
+		return ""
+	}
+
+	upstreamStatus, err := getUpstreamStatus(repo, branch)
+	if err != nil {
+		fmt.Printf("<ERROR: could not get status of upstream branch: %v>\n", err)
+		return ""
+	}
+
+	branchStatus := strings.Trim(fmt.Sprintf("%s, %s", parentStatus, upstreamStatus), ", ")
+	if branchStatus != "" {
+		branchInfo = fmt.Sprintf("(%s)", branchStatus)
+	}
+
+	if branch.PullRequest != nil && branch.PullRequest.Permalink != "" {
+		branchInfo = branch.PullRequest.Permalink + " " + branchInfo
+	}
+
+	return branchInfo
+}
+
+// Check if branch is up to date with the parent branch.
+func getParentStatus(repo *git.Repo, branch meta.Branch) (string, error) {
+	parentHead, err := repo.RevParse(&git.RevParse{Rev: branch.Parent.Name})
+	if err != nil {
+		return "", err
+	}
+
+	mergeBase, err := repo.MergeBase(&git.MergeBase{
+		Revs: []string{parentHead, branch.Name},
+	})
+	if err != nil {
+		return "", err
+	}
+	if mergeBase == parentHead {
+		return "", nil
+	}
+	
+	return "needs sync", nil
+}
+
+// Check if branch is up to date with the upstream branch.
+// This is doing `git diff <givenBranch> remotes/origin/<givenBranch>`
+func getUpstreamStatus(repo *git.Repo, branch meta.Branch) (string, error) {
+	upstreamExists, err := repo.DoesRemoteBranchExist(branch.Name)
```

Comment:
note that this only checks the local reference for if the upstream branch exists. also if the branch has already been merged and deleted, we will not display the `not pushed` messaged.

discussed offline with @draftcode , I think this should be ok for now.